doc/nrf/ug_fw_update.rst

```diff
@@ -0,0 +1,314 @@
+However, when working within the |NCS| framework, it is recommended to use the script that is included with the fork of MCUboot used by the |NCS|.
+
+See below for examples of imgtool used to create some commonly used key types:
+
+.. tabs::
+
+   .. group-tab:: ECDSA
+
+      .. code-block:: console
+
+         python3 bootlooader/mcuboot/scripts/imgtool.py keygen -t ecdsa-p256 -k priv.pem
+
+   .. group-tab:: RSA-2048
+
+      .. code-block:: console
+
+         python3 bootlooader/mcuboot/scripts/imgtool.py keygen -t rsa-2048 -k priv.pem
+
+For a full list of the supported types, use the ``--help`` argument with the tool or any of its commands.
+
+.. _ug_fw_update_keys_python:
+
+Using Python to generate keys
+-----------------------------
+
+The |NCS| ships with an internal script used for interacting with private/public keys with the |NSIB|.
+This script is maintained by the |NCS| team and utilizes the ``ecdsa`` package available from PyPI.
+As such, it is only valid for use with ECDSA keys.
+
+See below for an example of this internal Python script used to generate keys.
+
+.. code-block:: console
+
+   python3 nrf/scripts/bootloader/keygen.py --private -o priv.pem
+
+.. _ug_fw_update_development_keys:
+
+Using development keys
+======================
+
+When testing the bootloader chain, generating and using custom signing keys is optional.
+If you do not provide your own keys through Kconfig options, the build system will automatically create debug keys, depending on the bootloaders compiled into the application.
+
+.. caution::
+
+   Keys that are automatically used or generated by bootloaders for image signature validation are intended for *development or debug use only*.
+
+   Applications should *never* go into production when they are not protected by secure keys.
+   It is strongly encouraged to create and store these keys in a safe location, not only to protect the security of the application but also to ensure that the hardware can receive firmware updates throughout the project lifecycle.
+
+   Moreover, while the default keys for MCUboot are tracked in its repository, and are therefore publicly visible, the development/debug keys autogenerated by |NSIB| will change whenever the build directory is removed and rebuilt from scratch.
+   If you are also not programming the |NSIB| when this happens, relaying on the default ECDSA key to sign an application or an upgradable second-stage bootloader image will result in a failed boot chain validation.
+
+   You can avoid this issue by storing a custom private key outside of the build directory during development.
+
+.. _ug_fw_update_key_revocation:
+
+Revoking private keys
+=====================
+
+The |NSIB| supports the revocation of public signing keys that are available to validate the next image in the secure boot chain.
+Key revocation can be an useful security or backup tool for devices that have already been deployed to the field.
```

Comment:
"backup tool" gives slightly confusing connotations here IMO.

Reply:
Removed and tweaked the wording accordingly in latest commit.